### PR TITLE
Fix ctrl+[ not getting out of insert mode

### PR DESCRIPTION
**Pull request by liamnaddell**
This issue should be resolved with the following patch

### CODE REVIEW

File: src/keymap/vi.rs [hunk 1]
```diff
@@ -460,6 +460,7 @@ impl<'a, W: Write> Vi<'a, W> {
         if self.last_insert.is_some() {
             // leave insert mode
             try!(self.handle_key_core(Key::Esc));
+            try!(self.handle_key_core(Key::Ctrl('[')));
```
Comment (iamcodemaker) [inline src/keymap/vi.rs]: This should be removed.

File: src/keymap/vi.rs [hunk 2]
```diff
@@ -1198,6 +1199,32 @@ mod tests {
         assert_eq!(map.ed.cursor(), 0);
     }
 
+    #[test]
+    /// Cursor moves left when exiting insert mode.
+    fn vi_switch_from_insert_ctrl_lb() {
```
Comment (iamcodemaker) [inline src/keymap/vi.rs]: Instead of duplicating this test with `ctrl-[`, replace some (or all) of the `Esc` presses in the original test with `ctrl-[`.

**Review by iamcodemaker — CHANGES_REQUESTED** [review]

**Comment by liamnaddell** [comment]
I added a new commit bb84002 that should address your concerns

**Review by iamcodemaker — APPROVED** [review]

**Comment by liamnaddell** [comment]
Oops didn't mean to add the commit "Add upstream fork commits except 1" to this pr!" Sorry

**Pull request closed without merging**